src/and.js

```diff
@@ -19,5 +23,5 @@ var _curry2 = require('./internal/_curry2');
  *      R.and(null, ''); => null
  */
 module.exports = _curry2(function and(a, b) {
-  return a && b;
+  return _hasMethod('and', a) ? a.and(b) : a && b;
```

Comment:
interesting. `&&` is commutative, so it may not matter. But typically we dispatch on the last argument, except in cases where it's an operator, e.g. `ap`. Why did you opt to dispatch on `a` instead of `b`?